pkg/controller/nodeipam/node_ipam_controller.go

```diff
@@ -135,12 +140,17 @@ func NewNodeIpamController(
 			NodeCIDRMaskSizes:    nodeCIDRMaskSizes,
 		}
 
-		ic.cidrAllocator, err = ipam.New(kubeClient, cloud, nodeInformer, ic.allocatorType, allocatorParams)
+		ic.cidrAllocator, err = ipam.New(kubeClient, cloud, nodeInformer, clusterCIDRInformer, ic.allocatorType, allocatorParams)
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	if ic.allocatorType == ipam.MultiCIDRRangeAllocatorType {
```

Comment:
do we need to update the check on line 107?

```
	if allocatorType != ipam.CloudAllocatorType {
		if len(clusterCIDRs) == 0 {
			klog.Fatal("Controller: Must specify --cluster-cidr if --allocate-node-cidrs is set")
		}
```

Reply:
I think we need to determine if it is a required flag, another alternative that the users might take is to define a ClusterCIDR object

If --cluster-cidr we will create a default ClusterCIDR out of it, however I am not sure if it is a must